src/firebase/useFirebase.ts

```diff
@@ -0,0 +1,51 @@
+import {
+    DocumentData,
+    DocumentReference,
+    addDoc,
+    collection,
+    deleteDoc,
+    doc,
+    getDoc,
+    setDoc,
+} from "@firebase/firestore";
+import { db } from "./firebase";
+
+export const useFirebaseStorage = () => {
+
+    const getFirestoreData = async (
+        path: string,
+        id: string
+    ) => {
+        const item = await getDoc(doc(db, path, id))
+        if(!item.exists()) {
+            return null;
+        }
+        return item.data();
+    }
+
+    const setFirestoreData = async (
+        path: string,
+        data: DocumentData
+    ) => {
+        const item = await addDoc(collection(db, path), data);
+        return item;
+    }
```

Comment:
1. add types for all metods
```suggestion
const setFirestoreDataById = async <T extends DocumentData>(
  path: DatabasePaths,
  id: string,
  data: T
): Promise<DocumentReference<T>> => {
  const docRef = doc(db, path, id) as DocumentReference<T>;

  await setDoc(docRef, data);

  return docRef;
}
```

2. use your id so that we have more opportunities to create documents in the db
3. create an Enum for paths to avoid mistakes in the future when creating documents
5. do not create hooks for methods, use an Api file, since we will use Thunk for asynchronous requests in the db